sdk/verify.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package sdk
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net"
+
+	"github.com/edgelesssys/contrast/internal/atls"
+	"github.com/edgelesssys/contrast/internal/grpc/dialer"
+	"github.com/edgelesssys/contrast/internal/manifest"
+	"github.com/edgelesssys/contrast/internal/userapi"
+)
+
+// Client is used to interact with a Contrast deployment.
+type Client struct {
+	log *slog.Logger
+}
+
+// New returns a Client.
+func New(log *slog.Logger) Client {
+	return Client{
+		log: log,
+	}
+}
+
+// Verify checks if a given manifest is the latest manifest in the given history.
+func Verify(expected []byte, history [][]byte) error {
+	currentManifest := history[len(history)-1]
+	if !bytes.Equal(currentManifest, expected) {
+		return fmt.Errorf("active manifest does not match expected manifest")
+	}
+
+	return nil
+}
+
+// GetManifests calls GetManifests on the coordinator's userapi.
+func (c Client) GetManifests(ctx context.Context, manifestBytes []byte, endpoint string, policyHash []byte) (GetManifestsResponse, error) {
```

Comment:
the endpoint should be part of the client

Reply:
What do you mean here? It is intended that the sdk wraps the userapi pkg.

Reply:
We talked about this on the phone. Right now I don't see a good argument for putting the endpoint into the client. To me the endpoint seems as much or as little sth that should be provided on each call as e.g. the manifest. It also seemed closer to the CLI behavior. One provides the endpoint on each command, instead of writing it into a config file.

If more people feel like `endpoint` should be a Client property I will change it.